internal/host/plugin/repository_host_catalog.go

```diff
@@ -143,6 +146,279 @@ func (r *Repository) CreateCatalog(ctx context.Context, c *HostCatalog, _ ...Opt
+	}
+
+	var catalogsUpdated, secretsUpdated int
+	var returnedCatalog *HostCatalog
+	_, err = r.writer.DoTx(
+		ctx,
+		db.StdRetryCnt,
+		db.ExpBackoff{},
+		func(_ db.Reader, w db.Writer) error {
+			msgs := make([]*oplog.Message, 0, 3)
+			ticket, err := w.GetTicket(c)
+			if err != nil {
+				return errors.Wrap(ctx, err, op, errors.WithMsg("unable to get ticket"))
+			}
+
+			if len(dbMask) != 0 || len(nullFields) != 0 {
+				returnedCatalog = c.clone()
+				var cOplogMsg oplog.Message
+				catalogsUpdated, err = w.Update(
+					ctx,
+					returnedCatalog,
+					dbMask,
+					nullFields,
+					db.NewOplogMsg(&cOplogMsg),
+					db.WithVersion(&version),
+				)
+				if err != nil {
+					return errors.Wrap(ctx, err, op)
+				}
+				if catalogsUpdated > 1 {
+					return errors.New(ctx, errors.MultipleRecords, op, "more than 1 resource would have been updated")
+				}
+				msgs = append(msgs, &cOplogMsg)
+			} else {
+				// Returned catalog needs to be the old copy, as no fields in the
+				// catalog itself are being updated (note: secrets may still be
+				// updated).
+				returnedCatalog = currentCatalog.clone()
+			}
+
+			if hcSecret != nil {
+				if deleteSecrets {
+					// We didn't set/encrypt the persisted data because there was
+					// none returned. Just delete the entry.
+					deletedSecret := hcSecret.clone()
+					q, v := deletedSecret.deleteQuery()
+					var err error
+					secretsUpdated, err = w.Exec(ctx, q, v)
+					if err != nil {
+						return errors.Wrap(ctx, err, op)
+					}
+					if secretsUpdated != 1 {
+						return errors.New(ctx, errors.MultipleRecords, op, fmt.Sprintf("expected 1 catalog secret to be deleted, got %d", secretsUpdated))
+					}
+					msgs = append(msgs, deletedSecret.oplogMessage(db.DeleteOp))
+				} else {
+					// Update the secrets.
+					updatedSecret := hcSecret.clone()
+					q, v := updatedSecret.upsertQuery()
+					var err error
+					secretsUpdated, err = w.Exec(ctx, q, v)
+					if err != nil {
+						return errors.Wrap(ctx, err, op)
+					}
+					if secretsUpdated > 1 {
+						return errors.New(ctx, errors.MultipleRecords, op, "more than 1 catalog secret would have been updated")
+					}
+					msgs = append(msgs, updatedSecret.oplogMessage(db.UpdateOp))
+				}
+			}
+
+			metadata := c.oplog(oplog.OpType_OP_TYPE_UPDATE)
+			if err := w.WriteOplogEntryWith(ctx, oplogWrapper, ticket, metadata, msgs); err != nil {
+				return errors.Wrap(ctx, err, op, errors.WithMsg("unable to write oplog"))
+			}
+			return nil
+		},
+	)
+
+	if err != nil {
+		if errors.IsUniqueError(err) {
+			return nil, db.NoRowsAffected, db.NoRowsAffected, errors.Wrap(ctx, err, op, errors.WithMsg(fmt.Sprintf("in %s: name %s already exists", c.PublicId, c.Name)))
+		}
+		return nil, db.NoRowsAffected, db.NoRowsAffected, errors.Wrap(ctx, err, op, errors.WithMsg(fmt.Sprintf("in %s", c.PublicId)))
+	}
+
+	return returnedCatalog, catalogsUpdated, secretsUpdated, nil
```

Comment:
Don't separate out the catalogs updated or secrets updated.  Provide a single count for both of them.  A catalog whose secret gets updated should just count as a catalog which is updated.  A catalog whose secret and non-secret gets updated should still just count as a single catalog that got updated.

Reply:
Fixed!